python/ray/tests/test_global_state.py

```diff
@@ -154,6 +155,39 @@ def ready(self):
     )
 
 
+def test_node_table(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(node_name="head_node", include_dashboard=False)
+    head_context = ray.init(address=cluster.address, include_dashboard=False)
+    cluster.add_node(node_name="worker_node", include_dashboard=False)
+    cluster.wait_for_nodes()
```

Comment:
Can you also test the case where node_name is not given>? (and make sure it matches the expectation)